src/services/tag.service.js

```diff
@@ -61,6 +70,19 @@ export class TagService {
 
     await checkRecordByField(Tags, '_id', req.params.id, true);
 
+    if (parent_id == req.params.id) {
+      throw new ApiError(400, 'Parent tag cannot be itself');
+    }
+
+    const listChildrenTag = await Tags.find({ parent_id: req.params.id });
```

Comment:
Logic thì tôi thấy ko cần bàn nữa rồi. Tuy nhiên, tôi thấy vòng for ở đây sử dụng ko cần thiết cho lắm, chỗ này nếu là tôi thì tôi method some của js là ok rồi. Bạn xem rồi sửa lại nhé